src/controllers/sendMessage.js

```diff
@@ -3,78 +3,92 @@ import http from "http";
-      headers: {
-        "Content-Type": "application/json",
-        "Content-Length": Buffer.byteLength(body),
+      const postOptions = {
+        host: "127.0.0.1",
+        //host: "messageapp",
+        port: 3000,
+        path: "/message",
+        method: "post",
+        json: true,
+        headers: {
+          "Content-Type": "application/json",
+          "Content-Length": Buffer.byteLength(body),
+        }
       }
-    }
 
-    const postReq = http.request(postOptions);
-    postReq.on("response", async (postRes) => {
-      try {
-        await saveMessage({
-          ...req.body,
-          status: postRes.statusCode === 200 ? "OK" : "ERROR",
-        });
-        if (postRes.statusCode !== 200) {
-          throw new Error('Error in the messageapp request');
+      const postReq = http.request(postOptions);
+      postReq.on("response", async (postRes) => {
+        try {
+          await saveMessage({
+            ...req.body,
+            status: postRes.statusCode === 200 ? "OK" : "ERROR",
+            taskID: id
+            //should retry sending the message
+          });
+
+          res.statusCode = 200;
+          res.end(`your message was sent. ID:${id}`)
+
+        } catch (err) {
+          await deductCredit(-messageCost)
+          res.statusCode = 500;
+          res.end(`your message wasnt send, no credit deduction`)
+          //res.end(`Internal server error: SERVICE ERROR ${err}`);
         }
-        res.statusCode = 200;
-        res.end(postRes.body);
-      } catch (err) {
-        await deductCredit(-messageCost)
-        res.statusCode = 500;
-        res.end(`Internal server error: SERVICE ERROR ${err}`);
-      }
-    });
-
-    postReq.on("timeout", async () => {
-      console.error("Timeout Exceeded!");
-      postReq.abort();
-      try {
-        await saveMessage({
-          ...req.body,
-          status: "TIMEOUT",
-        });
-      } finally {
+      });
+
+      postReq.on("timeout", async () => {
+        console.error("Timeout Exceeded!");
+        postReq.abort();
+        try {
+          await saveMessage({
+            ...req.body,
+            status: "TIMEOUT",
+            taskID: id
+          });
+        } finally {
+          res.statusCode = 500;
+          res.end("Internal server error: TIMEOUT");
+        }
+      });
+
+      postReq.on("error", async (err) => {
+        console.log("ERRRooooooooooorrrrRRRRRRRRRR")
         res.statusCode = 500;
-        res.end("Internal server error: TIMEOUT");
-      }
-    });
-
-    postReq.on("error", async (err) => {
-      console.log("ERRRORAAAAZOOOO")
-      await deductCredit(-messageCost)
-      res.statusCode = 500;
-      res.json(err);
-    });
-    postReq.write(body);
-    postReq.end();
+        res.json(err);
+      });
+      postReq.write(body);
+      postReq.end();
+    }
+    const taskID = uniqid()
+    await deductCredit(messageCost)
+    mainQueue(messageSendingProcess(body, taskID), taskID)
+
```

Comment:
I would encapsulate this more, maybe into a jobs folder which exports each proccess to send to the queue (messageSendingProcess)